lib/repl.js

```diff
@@ -778,6 +779,19 @@ function filteredOwnPropertyNames(obj) {
   return Object.getOwnPropertyNames(obj).filter(intFilter);
 }
 
+function getGlobalLexicalScopeNames() {
+  inspector.open();
+  const session = new inspector.Session();
+  session.connect();
+  let names;
+  session.post('Runtime.globalLexicalScopeNames', {}, function(error, result) {
+    names = result.names;
+  });
+  session.disconnect();
+  inspector.close();
+  return names;
+}
```

Comment:
Wait – you can use the inspector to do synchronous exception of the running process? This is absolutely amazing?? :heart_eyes:

Reply:
Yes, it's cool. However in reality it's actually a [released Zalgo](http://blog.izs.me/post/59142742143/designing-apis-for-asynchrony):

- In some cases, the callback is called synchronously.
- In others (such as [`Runtime.awaitPromise`](https://chromedevtools.github.io/devtools-protocol/v8/Runtime/#method-awaitPromise) or [`Runtime.evaluate`](https://chromedevtools.github.io/devtools-protocol/v8/Runtime/#method-evaluate) with `awaitPromise` option specified) the callback is called asynchronously.

I guess either nobody thought of this problem when the API or added, or synchronous calling in those circumstances under which it's possible is too valuable to force asynchrony, that it remains this way.

Reply:
> or synchronous calling in those circumstances under which it's possible is too valuable to force asynchrony

I would go with that. ;)

Reply:
Yeah, many thanks to @ofrobots who explained this to me at NINA!